api/src/main/java/org/allaymc/api/entity/component/EntityBaseComponent.java

```diff
@@ -790,11 +806,45 @@ default boolean isEyesInWater() {
                eyesBlockState.getBlockStateData().computeOffsetShape(MathUtils.floor(eyeLoc)).intersectsPoint(eyeLoc);
     }
 
+    /**
+     * Check if the entity is a Boss mob.
+     *
+     * @return {@code true} if the entity is a Boss mob, otherwise {@code false}.
+     */
+    default boolean isBoss() {
+        return getEntityType().equals(EntityTypes.ENDER_DRAGON) ||
+               getEntityType().equals(EntityTypes.WITHER);
+    }
+
+    /**
+     * Check if the entity is an Undead mob.
+     *
+     * @return {@code true} if the entity is an Undead mob, otherwise {@code false}.
+     */
+    default boolean isUndead() {
```

Comment:
Find a better way to do that...? For example add an EntityUndeadComponent and use instanceof here